scripts/build/termux_download.sh

```diff
@@ -1,53 +1,68 @@
 #!/usr/bin/bash
 
 termux_download() {
-	if [ $# != 3 ]; then
+	if [[ $# != 3 ]]; then
 		termux_error_exit "termux_download(): Invalid arguments - expected \$URL \$DESTINATION \$CHECKSUM"
 	fi
 	local URL="$1"
 	local DESTINATION="$2"
 	local CHECKSUM="$3"
 
-	if [ -f "$DESTINATION" ] && [ "$CHECKSUM" != "SKIP_CHECKSUM" ]; then
+	if [[ -f "$DESTINATION" ]] && [[ "$CHECKSUM" != "SKIP_CHECKSUM" ]]; then
 		# Keep existing file if checksum matches.
 		local EXISTING_CHECKSUM
-		EXISTING_CHECKSUM=$(sha256sum "$DESTINATION" | cut -f 1 -d ' ')
-		if [ "$EXISTING_CHECKSUM" = "$CHECKSUM" ]; then return; fi
+		EXISTING_CHECKSUM=$(sha256sum "$DESTINATION" | cut -d' ' -f1)
+		[[ "$EXISTING_CHECKSUM" == "$CHECKSUM" ]] && return
 	fi
 
 	local TMPFILE
 	local -a CURL_OPTIONS=(
 		--fail
-		--retry 20
+		--retry 10
 		--retry-connrefused
-		--retry-delay 30
+		--retry-delay 5
+		--speed-limit 10000
+		--speed-time 30
 		--location
 	)
 	TMPFILE=$(mktemp "$TERMUX_PKG_TMPDIR/download.${TERMUX_PKG_NAME-unnamed}.XXXXXXXXX")
-	if [ "${TERMUX_QUIET_BUILD-}" = true ]; then
+	if [[ "${TERMUX_QUIET_BUILD-}" == "true" ]]; then
 		CURL_OPTIONS+=( --no-progress-meter)
 	fi
 
 	echo "Downloading ${URL}"
-	if curl "${CURL_OPTIONS[@]}" --output "$TMPFILE" "$URL"; then
-		local ACTUAL_CHECKSUM
-		ACTUAL_CHECKSUM=$(sha256sum "$TMPFILE" | cut -f 1 -d ' ')
-		if [ "$CHECKSUM" != "SKIP_CHECKSUM" ]; then
-			if [ "$CHECKSUM" != "$ACTUAL_CHECKSUM" ]; then
-				>&2 printf "Wrong checksum for %s\nExpected: %s\nActual:   %s\n" \
-					   "$URL" "$CHECKSUM" "$ACTUAL_CHECKSUM"
-				return 1
+	if ! curl "${CURL_OPTIONS[@]}" --output "$TMPFILE" "$URL"; then
+		local e=1
+		local retry=2
+		local try
+		for try in $(seq "${retry}"); do
+			echo "Retrying #${try} download ${URL}"
+			sleep 1
```

Comment:
Would be better to increase the sleep time to `60s` or `30s` at least. Previously, total retry time was `570s`/`9.5min`, where there was a wait of `30s` between each of the `20` tries. Now, the total retry time will be `45s`, where there is a wait of `5s` between each of the `10` tries, and then another retry period of `45s` after `1s`, so total `91s`/`1min31s`. It's very likely that sometimes servers will become inactive for longer than that. If we increase sleep time to `60s`, then total retry period would be `2min31s`, its still `~1/4` of the previous `10min`, so much lower with lower chance of build failure. The above calculations assume that the last retry wait doesn't happen as that would be after the last retry, and some time will be spent by `curl` on each retry until it fails/times out, so actual execution time would be longer.